crates/threshold-bls/src/group.rs

```diff
@@ -9,17 +9,17 @@ use std::marker::PhantomData;
 /// which is also equipped with a multiplication transformation.
 /// Two implementations are for Scalar which forms a ring so RHS is the same
 /// and Point which can be multiplied by a scalar of its prime field.
-pub trait Element<RHS = Self>:
-    Clone + Display + Debug + Eq + Serialize + for<'a> Deserialize<'a>
-{
+pub trait Element: Clone + Display + Debug + Eq + Serialize + for<'a> Deserialize<'a> {
```

Comment:
What is this thing doing `for<'a' Deserialize<'a>` ? lol never seen until now

Reply:
Higher Rank Trait bounds: https://doc.rust-lang.org/nomicon/hrtb.html

Serde example: https://journal.infinitenegativeutility.com/lifetime-quantification-and-higher-ranked-trait-bounds